File: python_converted/03_high_complexity/src/core/interfaces/types/weather.py
from typing import Any, Dict, List, Union



WeatherType = Union[, 'clear', 'rain', 'storm', 'snow', 'fog', 'sandstorm', 'heatwave', 'blizzard']
WeatherIntensity = Union['light', 'moderate', 'heavy', 'extreme']
class WeatherEffect:
    type: Union['visibility', 'movement', 'combat', 'status']
    value: float
    description: str
class WeatherState:
    type: WeatherType
    intensity: WeatherIntensity
    duration: float
    effects: List[WeatherEffect]
    visualEffects: Dict[str, Any]
}
class RegionWeather:
    current: \'WeatherState\'
    forecast: List[WeatherState]
    lastUpdated: float
class WeatherSystemState:
    regions: Dict[str, RegionWeather>
    globalEffects: List[WeatherEffect]
    transitionState?: {
    from: \'WeatherState\'
    to: \'WeatherState\'
    progress: float
}
class WeatherParams:
    region: str
    biome: Union['temperate', 'desert', 'tundra', 'coastal', 'tropical']
    season: Union['spring', 'summer', 'fall', 'winter']
    altitude: float
    forceWeather?: {
    type: WeatherType
    intensity: WeatherIntensity
}
class WeatherConfig:
    transitionDuration: float
    updateInterval: float
    forecastLength: float
    intensityThresholds: Dict[WeatherIntensity, float>
    seasonalWeights: Dict[str, Dict[WeatherType, float>>
    terrainWeights: Dict[str, Dict[WeatherType, float>> 

File: app/frontend/src/core/interfaces/types/weather.ts
// Weather system types and interfaces

export type WeatherType = 
  | 'clear'
  | 'rain'
  | 'storm'
  | 'snow'
  | 'fog'
  | 'sandstorm'
  | 'heatwave'
  | 'blizzard';

export type WeatherIntensity = 'light' | 'moderate' | 'heavy' | 'extreme';

export interface WeatherEffect {
  type: 'visibility' | 'movement' | 'combat' | 'status';
  value: number; // Modifier value
  description: string;
}

export interface WeatherState {
  type: WeatherType;
  intensity: WeatherIntensity;
  duration: number; // In game time units
  effects: WeatherEffect[];
  visualEffects: {
    particleSystem?: string;
    shader?: string;
    lighting?: {
      brightness: number;
      color: string;
    };
  };
}

export interface RegionWeather {
  current: WeatherState;
  forecast: WeatherState[]; // Future weather states
  lastUpdated: number; // Timestamp
}

export interface WeatherSystemState {
  regions: Record<string, RegionWeather>; // Region ID -> Weather data
  globalEffects: WeatherEffect[];
  transitionState?: {
    from: WeatherState;
    to: WeatherState;
    progress: number; // 0-1
  };
}

// Weather generation parameters
export interface WeatherParams {
  region: string;
  biome: 'temperate' | 'desert' | 'tundra' | 'coastal' | 'tropical';
  season: 'spring' | 'summer' | 'fall' | 'winter';
  altitude: number;
  forceWeather?: {
    type: WeatherType;
    intensity: WeatherIntensity;
  };
}

// Weather system configuration
export interface WeatherConfig {
  transitionDuration: number; // How long weather transitions take
  updateInterval: number; // How often to update weather
  forecastLength: number; // How many future weather states to generate
  intensityThresholds: Record<WeatherIntensity, number>;
  seasonalWeights: Record<string, Record<WeatherType, number>>;
  terrainWeights: Record<string, Record<WeatherType, number>>;
} 